src/Microsoft.Sbom.Api/Workflows/SBOMRedactionWorkflow.cs

```diff
@@ -17,17 +24,133 @@ public class SbomRedactionWorkflow : IWorkflow<SbomRedactionWorkflow>
 
     private readonly IConfiguration configuration;
 
+    private readonly IFileSystemUtils fileSystemUtils;
+
+    private readonly ValidatedSBOMFactory validatedSBOMFactory;
+
+    private readonly SbomRedactor sbomRedactor;
+
     public SbomRedactionWorkflow(
         ILogger log,
-        IConfiguration configuration)
+        IConfiguration configuration,
+        IFileSystemUtils fileSystemUtils,
+        ValidatedSBOMFactory validatedSBOMFactory = null,
+        SbomRedactor sbomRedactor = null)
     {
         this.log = log ?? throw new ArgumentNullException(nameof(log));
         this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.validatedSBOMFactory = validatedSBOMFactory ?? new ValidatedSBOMFactory();
+        this.sbomRedactor = sbomRedactor ?? new SbomRedactor();
     }
 
     public virtual async Task<bool> RunAsync()
     {
-        log.Information($"Running redaction for SBOM path {configuration.SbomPath?.Value} and SBOM dir {configuration.SbomDir?.Value}. Output dir: {configuration.OutputPath?.Value}");
-        return await Task.FromResult(true);
+        ValidateDirStrucutre();
+        var sboms = await GetValidSbomsAsync();
+
+        log.Information($"Running redaction on the following SBOMs: {string.Join(' ', sboms.Select(sbom => sbom.Key))}");
+        foreach (var (sbomPath, validatedSbom) in sboms)
+        {
+            try
+            {
+                var outputPath = GetOutputPath(sbomPath);
+                var redactedSpdx = await this.sbomRedactor.RedactSBOMAsync(validatedSbom);
+                using (var outStream = fileSystemUtils.OpenWrite(outputPath))
+                {
+                    await JsonSerializer.SerializeAsync(outStream, redactedSpdx);
+                }
+
+                log.Information($"Redacted SBOM {sbomPath} saved to {outputPath}");
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Failed to redact {sbomPath}: {ex.Message}", ex);
+            }
+        }
+
+        return true;
+    }
+
+    private async Task<IDictionary<string, IValidatedSBOM>> GetValidSbomsAsync()
```

Comment:
If the caller doesn't need to alter the contents, maybe
```suggestion
    private async Task<IReadOnlyDictionary<string, IValidatedSBOM>> GetValidSbomsAsync()
```